File: eSignature/examples/UnclaimedDraftCreateEmbeddedWithTemplate.rb
require "dropbox-sign"

Dropbox::Sign.configure do |config|
  # Configure HTTP basic authorization: api_key
  config.username = "YOUR_API_KEY"

  # or, configure Bearer (JWT) authorization: oauth2
  # config.access_token = "YOUR_ACCESS_TOKEN"
end

unclaimed_draft_api = Dropbox::Sign::UnclaimedDraftApi.new

signer_1 = Dropbox::Sign::SubUnclaimedDraftTemplateSigner.new
signer_1.role = "Client"
signer_1.name = "George"
signer_1.email_address = "george@example.com"

cc_1 = Dropbox::Sign::SubCC.new
cc_1.role = "Accounting"
cc_1.email_address = "accounting@example.com"

data = Dropbox::Sign::UnclaimedDraftCreateEmbeddedWithTemplateRequest.new
data.client_id = "ec64a202072370a737edf4a0eb7f4437"
data.template_ids = ["61a832ff0d8423f91d503e76bfbcc750f7417c78"]
data.requester_email_address = "jack@dropboxsign.com"
data.signers = [signer_1]
data.ccs = [cc_1]
data.test_mode = true

begin
  result = unclaimed_draft_api.unclaimed_draft_create_embedded_with_template(data)
  p result
rescue Dropbox::Sign::ApiError => e
  puts "Exception when calling Dropbox Sign API: #{e}"
end

File: eSignature/examples/UnclaimedDraftCreateEmbeddedWithTemplate.cs
using System;
using System.Collections.Generic;
using Dropbox.Sign.Api;
using Dropbox.Sign.Client;
using Dropbox.Sign.Model;

public class Example
{
    public static void Main()
    {
        var config = new Configuration();
        // Configure HTTP basic authorization: api_key
        config.Username = "YOUR_API_KEY";

        // or, configure Bearer (JWT) authorization: oauth2
        // config.AccessToken = "YOUR_BEARER_TOKEN";

        var unclaimedDraftApi = new UnclaimedDraftApi(config);

        var signer = new SubUnclaimedDraftTemplateSigner(
            role: "Client",
            name: "George",
            emailAddress: "george@example.com"
        );

        var cc1 = new SubCC(
            role: "Accounting",
            emailAddress: "accouting@email.com"
        );

        var data = new UnclaimedDraftCreateEmbeddedWithTemplateRequest(
            clientId: "1a659d9ad95bccd307ecad78d72192f8",
            templateIds: new List<string>(){"c26b8a16784a872da37ea946b9ddec7c1e11dff6"},
            requesterEmailAddress: "jack@dropboxsign.com",
            signers: new List<SubUnclaimedDraftTemplateSigner>(){signer},
            ccs: new List<SubCC>(){cc1},
            testMode: true
        );

        try
        {
            var result = unclaimedDraftApi.UnclaimedDraftCreateEmbeddedWithTemplate(data);
            Console.WriteLine(result);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling Dropbox Sign API: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}
